File: mobile/app/src/main/java/com/flightrisk/app/llm/NoOpLlmBackend.kt
package com.flightrisk.app.llm

import android.graphics.Bitmap

/**
 * Fallback LLM backend that returns "unavailable" for every request.
 *
 * Used when no real LLM backend is configured or reachable (e.g. the
 * device is on Tello WiFi with no internet, or no API key is set).
 * Always reports [isAvailable] = true so [LlmSelector] has a guaranteed
 * last-resort backend.
 */
class NoOpLlmBackend : LlmBackend {

    override val name: String = "none"

    override val isAvailable: Boolean = true

    override suspend fun analyzeMatch(
        referenceImage: Bitmap,
        candidateImage: Bitmap,
        description: String?,
    ): ReasoningResult = ReasoningResult(
        isMatch = false,
        confidence = "unavailable",
        reasoning = "LLM reasoning not available",
    )

    override suspend fun describeMatch(
        candidateImage: Bitmap,
        description: String,
    ): ReasoningResult = ReasoningResult(
        isMatch = false,
        confidence = "unavailable",
        reasoning = "LLM reasoning not available",
    )
}

File: ios/FlightRisk/LLM/NoOpLlmBackend.swift
import CoreGraphics

/// Fallback LLM backend that returns "unavailable" for every request.
///
/// Used when no real LLM backend is configured or reachable (e.g. the
/// device is on Tello WiFi with no internet, or no API key is set).
/// Always reports ``isAvailable`` as `true` so ``LlmSelector`` has a
/// guaranteed last-resort backend.
final class NoOpLlmBackend: LlmBackend {

    let name: String = "none"

    let isAvailable: Bool = true

    func analyzeMatch(
        referenceImage: CGImage,
        candidateImage: CGImage,
        description: String?
    ) async -> ReasoningResult {
        ReasoningResult(
            isMatch: false,
            confidence: "unavailable",
            reasoning: "LLM reasoning not available"
        )
    }

    func describeMatch(
        candidateImage: CGImage,
        description: String
    ) async -> ReasoningResult {
        ReasoningResult(
            isMatch: false,
            confidence: "unavailable",
            reasoning: "LLM reasoning not available"
        )
    }
}
